cli/src/main/kotlin/cli/function/Format.kt

```diff
@@ -1,10 +1,34 @@
 package cli.function
 
+import cli.util.generateAST
+import cli.util.getFilePath
+import formatter.FormattingRule
+import formatter.ProgramNodeFormatter
+import formatter.factory.FormatterMapFactory
+import lexer.Lexer
+import lexer.getTokenMap
+import parser.Parser
+import parser.factory.ProgramParserFactory
+import writer.FileWriter
+import writer.Writer
+
 /**
  * Formats the given source code.
  */
-class Format : CLIFunction {
+class Format(
+    private val lexer: Lexer = Lexer(getTokenMap()),
+    private val parser: Parser = ProgramParserFactory.create(),
+    private val writer: Writer = FileWriter(),
+) : CLIFunction {
+    private val configFile = "src/test/resources/format/config/formatter.test.config.json"
```

Comment:
Maybe I should pass it through the constructor ?

Reply:
I think it should be an argument from the map. Check the SCA function.